makefile

```diff
@@ -118,12 +117,17 @@ coveralls: lcov
 docs manual:
 	$(MAKE) -C doc/ $@ V=$(V)
 
-.PHONY: pre_gen
+.PHONY: pre_gen cmp
 pre_gen:
 	mkdir -p pre_gen
-	perl gen.pl
-	sed -e 's/[[:blank:]]*$$//' mpi.c > pre_gen/mpi.c
-	rm mpi.c
+	cat *mp_*.c > mp_all.c
```

Comment:
I would get rid of the pre_gen directory since it seems there will be only mp_all.c inside it.  This sed call seems also unnecessary? Only mp_all.c should be called differently such that helper.pl doesn't find it.

Reply:
Alternatively generate mp_all.c directly in pre_gen and not in ./

I sometimes had the issues with this stale file lying around and others had too I think.

Reply:
`ltm_amalgamated.c` fine?

Reply:
I would call it tommath_amalgam.c. This fits more with the rest.